libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -873,7 +873,13 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 					let arrayDataStart := <dataPosition>(array)
 					let deleteStart := add(arrayDataStart, newSlotCount)
 					let deleteEnd := add(arrayDataStart, oldSlotCount)
-					<clearStorageRange>(deleteStart, deleteEnd)
+					<?packed>
+						// if we are dealing with packed array and offset is greater than zero
+						// we have  to partially clear last slot that is still used, so decreasing start by one
+						let offset := mul(mod(newLen, <itemsPerSlot>), <storageBytes>)
+						if gt(offset, 0) { deleteStart := sub(deleteStart, 1) }
+					</packed>
+					<clearStorageRange>(deleteStart, <?packed>offset<!packed>0</packed>, deleteEnd)
```

Comment:
Wouldn't it be better to declare `let offset` right before `<?packed>` above and always use `offset` here?

Reply:
I was considering both options, and went with this one at the end. Will change as it is more clean